crates/store-dir/src/store_dir.rs

```diff
@@ -0,0 +1,91 @@
+use derive_more::From;
+use serde::{Deserialize, Serialize};
+use sha2::{digest, Sha512};
+use std::path::{self, PathBuf};
+
+/// Content hash of a file.
+pub type FileHash = digest::Output<Sha512>;
+
+/// Represent a store directory.
+///
+/// * The store directory stores all files that were acquired by installing packages with pacquet or pnpm.
+/// * The files in `node_modules` directories are hardlinks or reflinks to the files in the store directory.
+/// * The store directory can and often act as a global shared cache of all installation of different workspaces.
+/// * The location of the store directory can be customized by `store-dir` field.
+#[derive(Debug, PartialEq, Eq, From, Deserialize, Serialize)]
+#[serde(transparent)]
+pub struct StoreDir {
+    /// Path to the root of the store directory from which all sub-paths are derived.
+    ///
+    /// Consumer of this struct should interact with the sub-paths instead of this path.
+    root: PathBuf,
+}
+
+impl StoreDir {
+    /// Construct an instance of [`StoreDir`].
+    pub fn new(root: impl Into<PathBuf>) -> Self {
+        root.into().into()
+    }
+
+    /// Create an object that [displays](std::fmt::Display) the root of the store directory.
+    pub fn display(&self) -> path::Display {
+        self.root.display()
+    }
+
+    /// Get `{store}/v3`.
+    fn v3(&self) -> PathBuf {
```

Comment:
maybe just call it currentVersion?
```suggestion
    fn currentVersion(&self) -> PathBuf {
```
or maybe root can be the v3 directory.

Reply:
This functions returns the path to a sub dir named `v3`, not the version of the store.